services/cd-service/pkg/repository/repository.go

```diff
@@ -118,6 +123,16 @@ func openOrCreate(path string) (*git.Repository, error) {
 // Opens a repository. The repository is initialized and updated in the background.
 func New(ctx context.Context, cfg Config) (Repository, error) {
 	logger := logger.FromContext(ctx)
+
+	isMetricsEnabeld := ctx.Value("EnableMetrics")
+	if isMetricsEnabeld != nil && isMetricsEnabeld.(bool) {
```

Comment:
can't we read `cfg.EnableMetrics` here? Then we don't need to read it from the context.

Reply:
`cfg Config` here is the repo configuration, if I add EnableMetrics to repo configuration then it will work